federation/pkg/kubefed/init/init_test.go

```diff
@@ -141,9 +166,11 @@ func TestInitFederation(t *testing.T) {
 				t.Errorf("[%d] expected error: %s, got: %s, output: %s", i, tc.expectedErr, cmdErrMsg, buf.String())
 			}
 		}
-
-		testKubeconfigUpdate(t, tc.federation, tc.lbIP, tc.kubeconfigGlobal, tc.kubeconfigExplicit)
 	}
+
+	//test config update as a separate test case
+	tc := testCases[0]
+	testKubeconfigUpdate(t, tc.federation, tc.lbIP, tc.kubeconfigGlobal, tc.kubeconfigExplicit)
```

Comment:
Yeah, that was because there was only one test before this. The new tests were supposed to test variations. But it looks like the new ones don't. We should have tests that test different arguments to this function. Then it makes sense if it is in the loop. So please move it back.